server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -1507,7 +1540,7 @@ public static boolean hasPipeline(IndexRequest indexRequest) {
             || NOOP_PIPELINE_NAME.equals(indexRequest.getFinalPipeline()) == false;
     }
 
-    private record Pipelines(String defaultPipeline, String finalPipeline) {
+    public record Pipelines(String defaultPipeline, String finalPipeline) {
```

Comment:
In the spirit of https://github.com/elastic/elasticsearch/pull/116031/files#r1826153821, I'd like this PR a lot more if this record *didn't* become `public`.

Reply:
We're punting on this https://github.com/elastic/elasticsearch/pull/116031#discussion_r1826214460